nanoFramework.Storage/RemovableStorageEventManager.cs

```diff
@@ -0,0 +1,160 @@
+        /// you must associate the method with a <see cref="RemovableStorageDeviceEventHandler"/> delegate, and add this delegate to this event. 
+        /// </remarks>
+        public static event RemovableStorageDeviceEventHandler RemovableStorageDeviceInserted;
+
+        /// <summary>
+        /// Event that occurs when a Removable Storage Device is removed.
+        /// </summary>
+        /// <remarks>
+        /// The <see cref="RemovableStorageEventManager"/> class raises <see cref="RemovableStorageDeviceEventArgs"/> events when Removable Storage Devices (typically SD Cards and USB mass storage device) are inserted and removed.
+        /// 
+        /// To have a <see cref="RemovableStorageEventManager"/> object call an event-handling method when a <see cref="RemovableStorageDeviceRemoved"/> event occurs, 
+        /// you must associate the method with a <see cref="RemovableStorageDeviceEventHandler"/> delegate, and add this delegate to this event. 
+        /// </remarks>
+        public static event RemovableStorageDeviceEventHandler RemovableStorageDeviceRemoved;
+
+        static RemovableStorageEventManager()
+        {
+            RemovableStorageEventListener RemovableStorageEventListener = new RemovableStorageEventListener();
+
+            EventSink.AddEventProcessor(EventCategory.Storage, RemovableStorageEventListener);
+            EventSink.AddEventListener(EventCategory.Storage, RemovableStorageEventListener);
+        }
+
+        internal static void OnRemovableStorageEventCallback(RemovableStorageEvent removableStorageEvent)
+        {
+            switch (removableStorageEvent.EventType)
+            {
+                case RemovableStorageEventType.StorageDeviceInserted:
+                    {
+                        if (RemovableStorageDeviceInserted != null)
+                        {
+                            RemovableStorageDeviceEventArgs args = new RemovableStorageDeviceEventArgs(DriveIndexToPath(removableStorageEvent.DriveIndex), RemovableStorageDeviceEvent.Inserted);
+
+                            RemovableStorageDeviceInserted(null, args);
+                        }
+                        break;
+                    }
+                case RemovableStorageEventType.StorageDeviceRemoved:
+                    {
+                        if (RemovableStorageDeviceRemoved != null)
+                        {
+                            RemovableStorageDeviceEventArgs args = new RemovableStorageDeviceEventArgs(DriveIndexToPath(removableStorageEvent.DriveIndex), RemovableStorageDeviceEvent.Removed);
+
+                            RemovableStorageDeviceRemoved(null, args);
+                        }
+
+                        break;
+                    }
+                default:
+                    {
+                        break;
+                    }
+            }
+        }
+
+        internal static string DriveIndexToPath(byte driveIndex)
```

Comment:
I guess this part may not be needed or then better documented. Or at least contains somehow the non removable storage as well. Like C:
It's the moment to simplify all the complexity introduced by the Windows namespace

Reply:
I guess we could use the https://docs.microsoft.com/en-us/dotnet/api/system.io.driveinfo?view=net-5.0 class instead of this (I only learnt about it yesterday). That would also be more fitting with nanoframework/Home#799

Reply:
Yes, it makes sense. Still, we need the events as they don't seem to exist on any .NET namespaces...